osu.Game/Overlays/Chat/DrawableChannel.cs

```diff
@@ -112,19 +135,74 @@ private void pendingMessageResolved(Message existing, Message updated)
 
         private void messageRemoved(Message removed)
         {
-            ChatLineFlow.Children.FirstOrDefault(c => c.Message == removed)?.FadeColour(Color4.Red, 400).FadeOut(600).Expire();
+            getChatLines.FirstOrDefault(c => c.Message == removed)?.FadeColour(Color4.Red, 400).FadeOut(600).Expire();
         }
 
+        private IEnumerable<ChatLine> getChatLines => ChatLineFlow.Children.OfType<ChatLine>();
```

Comment:
Seems like a wrong name for a private property, although what do you think about returning the chat line container and apply the filtering inside:
```csharp
public IEnumerable<ChatLine> ChatLines => this.OfType<ChatLine>();
public IEnumerable<DaySeperator> Seperators => this.OfType<DaySeperator>();
```

Reply:
Another thing to note here this should be protected as the parent is.